README.md

```diff
@@ -5,17 +5,36 @@
 ## Running MiniMint locally
 MiniMint is tested and developed using rust `stable`, you can get it through your package manager or from [rustup.rs](https://rustup.rs/).
 
+MiniMint consists of three kinds of services:
+* federation member nodes (`server` binary) which make up the federation and run the consensus protocol among each other.
+* a Lightning gateway (`ln_gateway` binary) which acts as a bridge between the federation and the Lightning network allowing users to pay LN invoices with e-cash tokens.
+* user clients (`mint-client` binary) that interact with the federation nodes and the gateway
+
+In the following we will set up all three.
+
 ### Generating config
 You first need to generate some config. All scripts assume config to be located in a folder called `cfg`. Then you can generate the necessary configuration files as follows:
 
 ```shell
 mkdir -p cfg
-cargo run --bin configgen cfg <num_nodes> 5000 6000 <tier1> <tier2> …
+cargo run --bin configgen cfg <num_nodes> <federation_ports> <api_ports> <tier1> <tier2> …
+```
+
+The placeholders con be filled in as follows:
+* **`<num_nodes>`:** number of nodes to generate config for. Should be >= 4 and not too big as the cryptography of the BFT protocol is rather intense and you should ideally have 1 core per node.
+* **`<federation_ports>`:** base port for federation internal connections. If it is set to 5000 for example and there are 4 nodes they will use ports 5000, 5001, 5002 and 5003.
+* **`<api_ports>`:** base port for the federation node API server which user clients connect to. If it is set to 6000 for example and there are 4 nodes they will use ports 6000, 6001, 6002 and 6003.
+* **`<tier1> … <tier n>`:** E-cash token denominations/amount tiers in milli sat. There are different token denominations to increase efficiency so that instead of issuing 10 1sat tokens 1 10sat token can be issued. Generally powers of a base are a decent choice, e.g. powers of 10: 1 10 100 1000 10000 100000 1000000 10000000 100000000 1000000000 
```

Comment:
Off topic to this PR, but did you consider adding powers of 2 and powers of 3 (and multiples thereof) as well as a 1-2-5 series to the denominations?

Might help to make arbitrary amount payments with less tokens, with the cost of more denominations.

Reply:
You can add an example like `cargo run --bin configgen cfg 4 5000 6000 1 10 100 1000 10000 100000 1000000 10000000 100000000 1000000000`

Reply:
I mean, you can already do different powers afaik. Idk if any algorithm depends on divisibility between tiers, maybe the client but that could be fixed. For the config you can already define arbitrary denominations. Powers of x are most efficient imo when trading efficiency/privacy. 1, 2, 5 doesn't make much sense imo other than that is familiar to people (which doesn't matter given enough abstraction). I'd rather go with base 3.

Reply:
1-2-5 is useful when you send a rounded amount, it would be common if prices are denominated in Bitcoin, but I don't know if that is sufficient benefit to outweigh cost.

In any case, it's easy configurable, let's see how it gets used, thanks a lot!

Reply:
Thx for reviewing!

> Idk if any algorithm depends on divisibility between tiers

calls for more unit tests tbh, if anyone wants to tackle this @gourab337, @Maan2003, @NicolaLS (or anyone else) feel free to give it a shot :smiley: